src/transactions/ManageOfferOpFrameBase.cpp

```diff
@@ -367,7 +372,8 @@ ManageOfferOpFrameBase::doApply(AbstractLedgerTxn& ltxOuter)
         case ConvertResult::ePartial:
             sheepStays = true;
             break;
-        case ConvertResult::eFilterStop:
+        case ConvertResult::eFilterStopBadPrice:
+        case ConvertResult::eFilterStopCrossSelf:
```

Comment:
You should handle those two cases separately and remove the `if (!isResultSuccess())` (this was used to distinguish between `eFilterStopBadPrice` and `eFilterStopCrossSelf`), that way you can also move `setResultCrossSelf` into the `eFilterStopCrossSelf` case